File: js/problem3.js
function isItPrime(num) {
	var saveTime = Math.ceil(num /2) + 1;
	for (var i = 2; i < saveTime; i++) {
		if (num % i === 0) {
			return false;
		}
	}
	return true;
};

var primeFactors = function(num) {
	var array = [];
	findPrimeF(num);

	function findPrimeF(num) {
		if (isItPrime(num)) {
			return array.push(num);
		}
		for (var i = 2; i < num; i ++) {
			if (num % i === 0) {
				var n1 = i;
				var n2 = num / i;
				findPrimeF(n1);
				return findPrimeF(n2);
			}
		}
	};
	console.log(array);
	return array;
};

function largestPrimeFactor(array) {
	var largeToSmall = array.sort(function(a, b){return b-a});
	console.log(largeToSmall[0]);
	return largeToSmall[0];
};

largestPrimeFactor(primeFactors(224));
largestPrimeFactor(primeFactors(600851475143));

File: python/problem3.py
import math

def isItPrime(n):
	half = math.ceil(n / 2) + 1
	for i in range(2, half):
		if n % i == 0:
			return False
	return True


def prime_factors(num):
	pfs = []

	def find_prime_factors(num):
		if isItPrime(num):
			return pfs.append(num)
		for i in range(2, num):
			if num % i == 0:
				n1 = i
				n2 = num // i
				find_prime_factors(n1)
				return find_prime_factors(n2)


	find_prime_factors(num)
	return pfs


print(prime_factors(600851475143))
print(max(prime_factors(600851475143)))